integr8ly/integreatly-operator
# MGDAPI-4151 reduced log level message from error to warning for GetClusterType()

# Issue link
https://issues.redhat.com/browse/MGDAPI-4151

# What
New metrics were added in [MGDAPI-3695](https://issues.redhat.com/browse/MGDAPI-3695) one of which was getClusterType() however cluster value is only available on OpenShift v4.10 or better. 
This will prevent the installation from progressing because the cluster value doesn't exist on those versions of OpenShift. This PR changes the error to a warning to allow the installation to complete.

# Verification steps

1. Provision a cluster
2. Check out this [PR](https://github.com/integr8ly/integreatly-operator/pull/2769). 
3. Change [this line](https://github.com/integr8ly/integreatly-operator/blob/master/pkg/resources/cluster.go#L81) to 
`if tag.Key == "red-hat-clustertype-this-should-fail"`
4. Build and push operator image: `OLM_TYPE=managed-api-service INSTALLATION_TYPE=managed-api ORG={your quay repository} make image/build/push`
5. Update the [containerImage](https://github.com/integr8ly/integreatly-operator/blob/master/bundles/managed-api-service/1.24.0/manifests/managed-api-service.clusterserviceversion.yaml#L54) and [image](https://github.com/integr8ly/integreatly-operator/blob/master/bundles/managed-api-service/1.24.0/manifests/managed-api-service.clusterserviceversion.yaml#L540) fields and remove the [replaces](https://github.com/integr8ly/integreatly-operator/blob/master/bundles/managed-api-service/1.24.0/manifests/managed-api-service.clusterserviceversion.yaml#L740) value.
6. Build and push the bundle and index images: `OLM_TYPE=managed-api-service UPGRADE=false BUNDLE_VERSIONS=1.24.0 ORG={your-quay-org} make create/olm/bundle`
7. Install RHOAM using the images you just created once the cluster is ready.
8. Once the authentication stage has started, confirm that the error exists in the operator logs: `oc logs -l name=rhmi-operator --tail=-1 | grep red-hat-clustertype`.
9. Scale down the rhmi-operator deployment `oc scale --replicas=0 deployment/rhmi-operator`
10. Revert the change on [this line](https://github.com/integr8ly/integreatly-operator/blob/master/pkg/resources/cluster.go#L81).
11. Rebuild and push the operator image: `OLM_TYPE=managed-api-service INSTALLATION_TYPE=managed-api ORG={your quay repository} make image/build/push` 
12. Scale up the rhmi-operator deployment `oc scale --replicas=1 deployment/rhmi-operator`
13. Wait a few minutes and confirm that the error is now a warning: `oc logs -l name=rhmi-operator --tail=-1 | grep red-hat-clustertype`.
